indexing-service/src/main/java/io/druid/indexing/common/task/CompactionTask.java

```diff
@@ -304,7 +319,10 @@ private static DataSchema createDataSchema(
 
     // find granularity spec
     // set rollup only if rollup is set for all segments
-    final boolean rollup = queryableIndices.stream().allMatch(index -> index.getMetadata().isRollup());
+    final boolean rollup = queryableIndices.stream().allMatch(index -> {
+      final Boolean isRollup = index.getMetadata().isRollup(); // We have already done null check
```

Comment:
Should probably say `// We have already done null check on index.getMetadata()` (since Rollup could still be null, and the comment is vague about that)

Reply:
Changed to `// We have already checked getMetadata() doesn't return null`.

Reply:
Even better!